pkg/daemon/ceph/osd/volume.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	batchArgs := append(baseArgs, []string{
+		osdsPerDeviceFlag,
+		strconv.Itoa(a.storeConfig.OSDsPerDevice),
+	}...)
+
+	configured := 0
+	for name, device := range devices.Entries {
+		if device.LegacyPartitionsFound {
+			logger.Infof("skipping device %s configured with legacy rook osd", name)
+			continue
+		}
+
+		if device.Data == -1 {
+			logger.Infof("configuring new device %s", name)
+			deviceArg := path.Join("/dev", name)
+			if device.Config.OSDsPerDevice <= 1 {
+				// the device will be configured as a batch at the end of the method
+				batchArgs = append(batchArgs, deviceArg)
+				configured++
+			} else {
+				// execute ceph-volume immediately with the device-specific setting instead of batching up multiple devices together
+				immediateExecuteArgs := append(baseArgs, []string{
+					deviceArg,
+					osdsPerDeviceFlag,
+					strconv.Itoa(device.Config.OSDsPerDevice),
+				}...)
+
+				if err := context.Executor.ExecuteCommand(false, "", cephVolumeCmd, immediateExecuteArgs...); err != nil {
+					return fmt.Errorf("failed ceph-volume. %+v", err)
+				}
+
+			}
+		} else {
+			logger.Infof("skipping device %s with osd %d already configured", name, device.Data)
+		}
+	}
+
+	if configured > 0 {
+		if err := context.Executor.ExecuteCommand(false, "", cephVolumeCmd, batchArgs...); err != nil {
+			return fmt.Errorf("failed ceph-volume. %+v", err)
+		}
+	}
+
+	return nil
+}
+func getCephVolumeSupported(context *clusterd.Context) (bool, error) {
+	_, err := context.Executor.ExecuteCommandWithOutput(false, "", cephVolumeCmd, "lvm", "batch", "--prepare")
+	if err != nil {
+		if cmdErr, ok := err.(*exec.CommandError); ok {
+			exitStatus := cmdErr.ExitStatus()
+			if exitStatus == int(syscall.ENOENT) {
+				logger.Infof("supported version of ceph-volume not available")
+				return false, nil
+			}
+			logger.Warningf("unknown return code from ceph-volume when checking for compatibility: %d", exitStatus)
+		}
+		logger.Warningf("unknown ceph-volume failure. %+v", err)
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func getCephVolumeOSDs(context *clusterd.Context, clusterName string) ([]oposd.OSDInfo, error) {
+	result, err := context.Executor.ExecuteCommandWithOutput(false, "", cephVolumeCmd, "lvm", "list", "--format", "json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve ceph-volume results. %+v", err)
+	}
+
+	var cephVolumeResult map[string][]osdInfo
+	err = json.Unmarshal([]byte(result), &cephVolumeResult)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve ceph-volume results. %+v", err)
+	}
+
+	var osds []oposd.OSDInfo
+	for name, osdInfo := range cephVolumeResult {
+		id, err := strconv.Atoi(name)
+		if err != nil {
+			logger.Errorf("bad osd returned from ceph-volume: %s", name)
+			continue
+		}
+		var osdFSID string
+		isFilestore := false
+		for _, osd := range osdInfo {
+			osdFSID = osd.Tags.OSDFSID
+			if strings.HasPrefix(osd.Path, "/dev/ceph-filestore") {
```

Comment:
I think I'm missing something here: `/dev/ceph-filestore`?

Reply:
I'd search for the `type` key and then figure out if it's bluestore or filestore.

Reply:
thanks, i'll look for the `type` key. This was not a great way to check for filestore.